### PR TITLE
added PointCloudAdapter specialization

### PR DESCRIPTION
**Pull request by startcode**
Previous implementation could not correctly distinguish ros message and
protobuf message

### CODE REVIEW

File: modules/common/adapters/point_cloud_adapter.h [hunk 1]
```diff
@@ -0,0 +1,304 @@
+/******************************************************************************
```
Comment (siyangy) [inline modules/common/adapters/point_cloud_adapter.h]: This almost copies adapter.h, what's the reason to have this duplication?

**Review by siyangy — CHANGES_REQUESTED** [review]

File: modules/common/adapters/point_cloud_adapter.h (hunk 1 above)
Comment (startcode) [reply modules/common/adapters/point_cloud_adapter.h]: previous impl. failed to dustinguish the messages.

Comment (siyangy) [reply modules/common/adapters/point_cloud_adapter.h]: can we hold off this and discuss it tomorrow? I'd like to avoid duplicate code if possible.

Comment (startcode) [reply modules/common/adapters/point_cloud_adapter.h]: sure

**Comment by startcode** [comment]
@siyangy updated. please review again.

**Review by siyangy — APPROVED** [review]

**Pull request merged**